samcli/lib/build/app_builder.py

```diff
@@ -147,29 +147,37 @@ def update_template(self, template_dict, original_template_path, built_artifacts
             Updated template
         """
 
-        original_dir = os.path.dirname(original_template_path)
+        original_dir = pathlib.Path(os.path.dirname(original_template_path))
 
         for logical_id, resource in template_dict.get("Resources", {}).items():
 
             if logical_id not in built_artifacts:
                 # this resource was not built. So skip it
                 continue
 
+            artifact_dir = pathlib.Path(built_artifacts[logical_id])
```

Comment:
is `built_artifacts[logical_id]` a full path here and will that path always exist?

Reply:
Added resolve in case it's not full path. 
Path exists since _build_layer and _build_function mkdir for each artifact_dir.